front_end/src/app/(main)/(tournaments)/tournaments/components/tournaments_search.tsx

```diff
@@ -0,0 +1,46 @@
+import { ChangeEvent, useEffect, useState } from "react";
+
+import ExpandableSearchInput from "@/components/expandable_search_input";
+import useSearchInputState from "@/hooks/use_search_input_state";
+
+import { TOURNAMENTS_SEARCH } from "../constants/query_params";
+
+const TournamentsSearch: React.FC = () => {
+  const [searchQuery, setSearchQuery] = useSearchInputState(
+    TOURNAMENTS_SEARCH,
+    {
+      mode: "client",
+      debounceTime: 300,
+      modifySearchParams: true,
+    }
+  );
+
+  const [draftQuery, setDraftQuery] = useState(searchQuery);
+  useEffect(() => setDraftQuery(searchQuery), [searchQuery]);
```

Comment:
It's not clear why the extra draftQuery state/setState is needed here. If that's really needed, can you leave a small comment to clarify? Although, at first sight, it looks like it's not needed.